fhir-persistence-jdbc/src/main/java/org/linuxforhealth/fhir/persistence/jdbc/util/NewQueryBuilder.java

```diff
@@ -227,12 +227,55 @@ public Select buildCountQuery(Class<?> resourceType, FHIRSearchContext searchCon
      * @return
      */
     private Select renderQuery(SearchQuery domainModel, FHIRSearchContext searchContext, SchemaType schemaType) throws FHIRPersistenceException {
-        final int offset = (searchContext.getPageNumber()-1) * searchContext.getPageSize();
-        final int rowsPerPage = searchContext.getPageSize();
+        // adjust the offset by -1 of the search is for any other page than the first page.
+        int offsetIncrement = getOffsetCounter(searchContext);
```

Comment:
I think using the term `increment` might be confusing because the value is negative. Perhaps better just to use `offset` or `firstRowOffset`

Reply:
This has been fixed as suggested.